internal/coreinternal/timeutils/parser.go

```diff
@@ -0,0 +1,98 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package timeutils // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/timeutils"
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	strptime "github.com/observiq/ctimefmt"
+)
+
+func StrptimeToGo(layout string) (string, error) {
+	return strptime.ToNative(layout)
+}
+
+func GetLocation(location *string, layout *string) (*time.Location, error) {
+	if location != nil && *location != "" {
+		// If "location" is specified, it must be in the local timezone database
```

Comment:
```suggestion
		// If location is specified, it must be in the local timezone database
```
Just a thought, feel free to ignore.